rpc/src/v1/impls/eth.rs

```diff
@@ -456,15 +447,15 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM> Eth for EthClient<C, SN, S, M, EM> where
 			None => Err(errors::state_pruned()),
 		};
 
-		future::done(res).boxed()
+		Box::new(future::done(res))
 	}
 
 	fn block_by_hash(&self, hash: RpcH256, include_txs: bool) -> BoxFuture<Option<RichBlock>, Error> {
-		future::done(self.block(BlockId::Hash(hash.into()), include_txs)).boxed()
+		Box::new(future::done(self.block(BlockId::Hash(hash.into()), include_txs)))
 	}
 
 	fn block_by_number(&self, num: BlockNumber, include_txs: bool) -> BoxFuture<Option<RichBlock>, Error> {
-		future::done(self.block(num.into(), include_txs)).boxed()
```

Comment:
why the change from `.boxed()` to `Box::new`?

Reply:
`boxed()` has been deprecated. https://github.com/alexcrichton/futures-rs/issues/228

Reply:
The motivation is that `.boxed()` requires `Future + Send + 'static` and sometimes it's not needed. Wrapping with `Box::new` covers both cases.

Reply:
And usage of boxed causes warning during the compilation. It litters the build log significantly